lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c

```diff
@@ -986,9 +986,9 @@ void mme_app_handle_delete_session_rsp(
        Do not send UE Context Release Command to eNB before receiving SGs IMSI
        Detach Ack from MSC/VLR */
     if (ue_context_p->sgs_context != NULL) {
-      if (((ue_context_p->sgs_detach_type !=
+      if (((ue_context_p->sgs_detach_type ==
```

Comment:
It's not enough to switch the equality check here, we also need to flip the logic. In this case,

```
if ((ue_context_p->sgs_detach_type == SGS_EXPLICIT_UE_INITIATED_IMSI_DETACH_FROM_NONEPS) ||
           (ue_context_p->sgs_detach_type == SGS_COMBINED_UE_INITIATED_IMSI_DETACH_FROM_EPS_N_NONEPS)) {
           OAILOG_FUNC_OUT(LOG_MME_APP);
 }
 if (ue_context_p->sgs_context->ts9_timer.id == MME_APP_TIMER_INACTIVE_ID) {
 mme_app_itti_ue_context_release(
            ue_context_p, ue_context_p->ue_context_rel_cause);
        ue_context_p->ue_context_rel_cause = S1AP_INVALID_CAUSE;
   }
```

Reply:
integ_test failing for me, but when I wipe everything (vagrant destroy {magma, magma_test}) and re-test on master they fail in the same way...